### ISSUE TITLE
Unicode support.

### ISSUE THREAD

**Issue by Jookia** [garrynewman/GWEN#3]
(Haha yes of course I would talk about this, but this has actually affect people's decisions in using GWEN.)

This isn't one specific issue, but kind of a few but they revolve around a few decisions and implementations.
### wchar_t and char usage

So last time I talked about this, I've been an avid wchar_t hater. I still don't use it, but GWEN is designed around it, so it's all fine. What does bother me is how the API uses a mix of wchar_t and char. For instance, in Platform.h we have a mix of String& and UnicodeString&. We have Unicode clipboard text (woo!) but no Unicode window titles. It's not consistent.

I'd suggest moving entirely to wchar_t, and converting internally. The only place I'd see useful for char is UTF-8 APIs like SDL, Allegro which take char\* for all their stuff, which is internal.
The alternative is using entirely char UTF-8, which sucks on Windows, but I have heard that BlackPhoenix has an entire UTF-8 port of GWEN.
### Current conversion code.

Simply put: The current conversion code is broken. It's easily fixable by pulling in a header-only library and using that. This actually affects things right now like Allegro's UTF-8 API not being used properly when drawing text.
